### DISCUSSION TITLE
Distinguish between Wrong Password and Non-Existent User on Login.

### DISCUSSION THREAD

**Discussion by FabrizioG202** [supabase/supabase#10724] [General]
I am trying to implement the basic authentication flow with the Flutter implementation of Supabase. I am logging in an user with `.signInWithPassword`.

I was wondering whether it was possible to distinguish between the situation in which an user tries to login with a wrong password and the situation in which an user tries to log-in with an email which is not recorded in the database. With Firebase, the login function throws 2 different kinds of errors in this situation but in supabase I just get the error `AuthException(message: Invalid login credentials, statusCode: 400)` in both cases. 

Is it possible to distinguish between them?

**Comment by dshukertjr** [comment]
I believe not. 

I think Supabase auth is designed not to expose whether it's just a wrong password, or the user does not exist to prevent attackers to just send bunch of emails to obtain a list of users of your service.

**Comment by JeremiahJacob261** [comment]
You can just use this little workaround to detect.
Table users is a table where all users data are stored in the database after sign up
 if(error.message === 'Invalid login credentials'){
            const { count, error } = await supabase
            .from('users')
            .select('*', { count: 'exact', head: true })
            .eq('email', emailer)
          console.log(count);
          if (count > 0) {
            alert('in corect password')
          }else{
            alert('non existing users')
          }
          }
